### PR TITLE
test: add tests for `w3 up` and `w3 ls` commands

### PR DESCRIPTION
**Pull request by alanshaw**
* Configure service through environment variables
* Add helpers for mocking ucanto services
* Mock upload and access ucanto APIs in tests
* Mock S3 bucket externally from tests (since response does not need to vary per test)
* Add tests for `w3 up` and `w3 ls`

### CODE REVIEW

File: lib.js [hunk 1]
```diff
@@ -39,3 +47,42 @@ export function unwarnify () {
...
+ */
+export function getClient () {
+  const store = new StoreConf({ profile: process.env.W3_STORE_NAME ?? 'w3cli' })
+
+  let serviceConf
+  if (
+    process.env.W3_ACCESS_SERVICE_DID &&
+    process.env.W3_ACCESS_SERVICE_URL &&
+    process.env.W3_UPLOAD_SERVICE_DID &&
+    process.env.W3_UPLOAD_SERVICE_URL
```
Comment (olizilla) [inline lib.js:62]: I'm hoping that with did-web we can simplify this down to just one string per service

File: test/bin.spec.js [hunk 2]
```diff
@@ -13,6 +34,94 @@ test('w3 --version', (t) => {
...
+
+  const env = createEnv(serviceSigner, t.context.serverURL)
+
+  const list0 = await execa('./bin.js', ['ls'], { env })
+  t.regex(list0.stdout, /No uploads in space/)
+
+  await execa('./bin.js', ['up', 'test/fixtures/pinpie.jpg'], { env })
+
+  const list1 = await execa('./bin.js', ['ls'], { env })
+  t.notThrows(() => CID.parse(list1.stdout.trim()))
```
Comment (olizilla) [inline test/bin.spec.js:126]: Might be better to pass --json here. We might change human friendly output of the ls command, but the (nd)json form is less likely to change.

**Review by olizilla — APPROVED** [review]
Note that you are drifting towards a zero-deps approach that doesn't hide any details. Previously we's have used `nock` for stubbing out http requests or `smoke` for mock http end points, and something like sinon for call counting. 

i'm not against showing all the details, but if we stick to commonly used deps then PRs like this would be quicker for others to absorb. No action, just flagging.

File: test/bin.spec.js (hunk 2 above)
Comment (alanshaw) [reply test/bin.spec.js:126]: ...or at least we're more likely to keep it backwards compatible

**Pull request merged**